rasterio/crs.pyx

```diff
@@ -473,6 +473,8 @@ cdef class CRS:
             return "OGC", matches["OGC"][0]
         elif "ESRI" in matches:
             return "ESRI", matches["ESRI"][0]
+        elif "IAU_2015" in matches:
+            return "IAU", matches["IAU_2015"][0]
```

Comment:
Don't we want to keep the official authority name?

```suggestion
            return "IAU_2015", matches["IAU_2015"][0]
```

Reply:
(side comment / FYI: to be honest IAU_2015 is more a PROJ hack that something really clean. Normally this should be a versionned code like "IAU:2015:XXXXX" (which PROJ also supports on ingestion), but for various practical reasons, when outputing stuff, it is more convenient for PROJ to behind together the authority name and version under "IAU_2015". The magic is done in https://github.com/OSGeo/PROJ/blob/de73f2416a2ab3f945a4c53467ebf0a3f201a183/data/sql/customizations_early.sql#L8C14-L8C35)

Reply:
It looks like the IAU is the only currently versioned authority in Proj (https://github.com/OSGeo/PROJ/blob/master/data/sql/versioned_auth_name_mapping.sql) and the codes are recorded with "IAU_2015" as the authority name (https://github.com/OSGeo/PROJ/blob/de73f2416a2ab3f945a4c53467ebf0a3f201a183/data/sql/proj_db_table_defs.sql#L237). 

So technically "IAU_2015" **is** the actual name of the authority according to GDAL/OGR/PROJ.

Using "IAU" here wouldn't allow the tests to pass as the `in` check looks for the whole key and not a partial, otherwise you'd need to do an awkward `'IAU' in key for key in matches.keys()` check described in the resolved comments above. However using just `IAU` as the authority does work when constructing CRSs as proj maintains the authority name mapping/there are other references to checks on this I see in GDAL/OGR that allows it to work in that direction.

Maybe @rouault knows more of the history for this, or could update OSRGetAuthorityName to not include the version/ add a new function that returns the version as a separate return variable? 

In any case I suspect this would take a new release of GDAL/OGR and I don't want to block progress on other projects to wait for that, so I'd like to move forward with the PR as-is currently.

Alternatively @sgillies we could remove the comparison check to authority names as I suggested in #3217 but that could result in a change in behavior that would not prioritize the EPSG authority over others. But I'd agree that'd be cleaner and more correct behavior in the end and then wouldn't require any delays. But I hadn't heard back on your opinion on that concept yet.

Reply:
> Maybe @rouault knows more of the history for this, or could update OSRGetAuthorityName to not include the version/ add a new function that returns the version as a separate return variable?

Maybe. I tried to not over-engineer stuff. We'll see what to do if we have a IAU_20xx to handle at some point.

My own position: "IAU_2015" as the PROJ/GDAL/rasterio/QGIS/whatever authority name is good enough for now.

Reply:
heh was typing the above when I think @rouault was replying and didn't see the update